packages/in_app_purchase/lib/src/in_app_purchase/purchase_details.dart

```diff
@@ -122,7 +126,23 @@ class PurchaseDetails {
   final String transactionDate;
 
   /// The status that this [PurchaseDetails] is currently on.
-  PurchaseStatus status;
+  PurchaseStatus get status => _status;
+  set status(PurchaseStatus status) {
+    if (_platform == _kPlatformIOS) {
+      if (status == PurchaseStatus.purchased ||
+          status == PurchaseStatus.error) {
+        pendingCompletePurchase = true;
+      }
+    }
+    if (_platform == _kPlatformAndroid) {
+      if (status == PurchaseStatus.purchased) {
```

Comment:
So does the user need to special case whether or not they call `completePurchase` based on what the status is _and_ whether they're on iOS or Android now, where with Android they should avoid calling it for `PurchaseStatus.error`? I think this should probably be mentioned in the `completePurchase` docs.

Reply:
see the response on https://github.com/flutter/plugins/pull/2287/files#r349738652

Reply:
Sorry, that link doesn't work anymore. I think the PR may be too big. Do you mind copy/pasting it?